modules/EvseManager/EvseManager.cpp

```diff
@@ -29,6 +29,22 @@ void EvseManager::init() {
     if (!(config.ac_hlc_enabled || config.ac_with_soc || config.charge_mode == "DC")) {
         slac_enabled = false;
     }
+
+    // Use SLAC MAC address for Autocharge if configured.
+    if (config.autocharge_use_slac_instead_of_hlc && slac_enabled) {
+        r_slac[0]->subscribe_ev_mac_address([this](const std::string& _token) {
+            types::authorization::ProvidedIdToken autocharge_token;
+
+            std::string token = _token;
+            token.erase(remove(token.begin(), token.end(), ':'), token.end());
+            autocharge_token.id_token = "VID:" + token;
```

Comment:
you could also write a factory function like
```
inline static types::authorization::ProvidedIdToken create_autocharge_token(const std::string& token) {
  types::authorization::ProvidedIdToken autocharge_token;
  autocharge_token.type = types::authorization::TokenType::Autocharge;
  autocharge_token.id_token = "VID:" + trim_colons_from_string(token);
  return autocharge_token;
}
```
this would be just more explicit

Reply:
done